src/transformers/models/cpmant/modeling_cpmant.py

```diff
@@ -0,0 +1,1475 @@
+        length: torch.Tensor,
+        context: torch.Tensor,
+        position: torch.Tensor,
+        segment: torch.Tensor,
+        span: torch.Tensor,
+        past_key_values=None,
+    ):
+        """
+        Args:
+            input (`torch.Tensor`): tokenized ids, shape = `(batch, seq_len)`
+            length (`torch.Tensor`): length of input, shape = `(batch)`
+            context (`torch.Tensor`): context determines whether model predicts, shape = `(batch, seq_len)`
+            position (`torch.Tensor`): position of input, shape = `(batch, seq_len)`
+            segment (`torch.Tensor`): segment of input, shape = `(batch, seq_len)`
+            span (`torch.Tensor`): span the context of input, shape = `(batch, seq_len)`
+            past_key_value (`Tuple(torch.FloatTensor)`, *optional*): cached past key and value projection states
+        """
+        batch = input.size(0)
+
+        if past_key_values is None:
+            past_length = 0
+            past_key_values = tuple([None] * self.encoder.num_layers)
+            input_prompt = input[:, : self.prompt_length].contiguous()
+            input_ids = input[:, self.prompt_length :].contiguous()
+
+            prompt_states = self.prompt_embedding(input_prompt)
+            hidden_states = self.input_embedding(input_ids)
+            segment_states = self.segment_embedding(segment)
+            hidden_states = torch.cat([prompt_states, hidden_states], 1) + segment_states
+
+        else:
+            past_length = past_key_values[0][0].size(-2)
+            segment_states = self.segment_embedding(segment)
+            hidden_states = self.input_embedding(input) + segment_states[:, -1:, :]
+
+        seqlen = past_length + input.size(1)
+
+        with torch.no_grad():
+            device = input.device
+            directional_mask_2d = torch.arange(seqlen, device=device) <= torch.arange(seqlen, device=device).view(
+                -1, 1
+            )
+            attention_mask = context[:, None, :] | (
+                context[:, :, None].logical_not() & directional_mask_2d.view(1, seqlen, seqlen)
+            )
+            attention_mask = attention_mask & (span[:, None, :] == span[:, :, None])
+            # mask for left paddding
+            mask_1d = (
+                torch.tensor(list(range(seqlen))[::-1], device=device)[None, :].repeat(batch, 1) < length[:, None]
+            )
+            attention_mask = mask_1d.view(batch, seqlen, 1) & mask_1d.view(batch, 1, seqlen) & attention_mask
+
+        position_bias = self.position_bias(position, position, segment, segment)
+
+        attention_mask = attention_mask[:, past_length:, :]
+        position_bias = position_bias[:, :, past_length:, :]
+
+        hidden_states, present_key_values = self.encoder(
+            hidden_states, attention_mask, position_bias, True, past_key_values
+        )
+        logits = self.lm_head(hidden_states)
+        return logits, hidden_states, present_key_values
```

Comment:
Can you return a dataclass object instead? For example: https://github.com/huggingface/transformers/blob/8fb4d0e4b46282d96386c229b9fb18bf7c80c25a/src/transformers/models/gpt_neox_japanese/modeling_gpt_neox_japanese.py#L697 or you can also define your own class